### PR TITLE
3 your first ticket a fancy button peter and teddy

### PR DESCRIPTION
**Pull request by pmorganelli**
finished button!!!!!

### CODE REVIEW

File: src/components/button5.tsx [hunk 1]
```diff
@@ -1,5 +1,41 @@
...
+    };
+    
+    return (
+        <div>
+            <button onClick={getWeather} className="bg-red-700 border-[6px] border-red-900  text-white px-12 py-4 rounded-full text-3xl font-bold">{props.label}</button>
+
+            {weather ? (
+                weather.probabilityOfPrecipitation.value > 55 ? (
+                    <div>
+                        <p className="text-center font-bold">Dream harder. We are in New England. {weather.probabilityOfPrecipitation.value}% chance of rain today.</p>
```
Comment (liam-strand) [inline src/components/button5.tsx:27]: My system is getting angry at the `weather.probabilityOfPrecipitation.value` line, saying that `weather.probabilityOfPrecipitation` doesn't have a member `value`. It is declared as a `string[]` on line 10, after all. 

I also am not seeing any number on the button, just the percent sign.

Maybe worth taking another look? Make sure you are running this with our version of node? It is also possible there is an issue on my end

File: src/components/button5.tsx [hunk 2]
```diff
@@ -1,5 +1,41 @@
...
+                    </div>
+                ) : (
+                    <div>
+                        <p className="text-center font-bold">Your wish has been granted. {weather.probabilityOfPrecipitation.value}% chance of rain today.</p>
```
Comment (liam-strand) [inline src/components/button5.tsx:31]: I love the snark :)

**Review by liam-strand — CHANGES_REQUESTED** [review]

File: src/components/button5.tsx (hunk 1 above)
Comment (liam-strand) [reply src/components/button5.tsx:27]: Ah i looked a little closer, it looks like the `value` could be null, which needs to be handled specially.

**Comment by teddyh-io** [comment]
How could the value even be null? Does that mean 0% chance of rain? 🤣

**Comment by pmorganelli** [comment]
Yeah we were talking about this yesterday -- what should we do in this case?

**Comment by liam-strand** [comment]
I'd take `null` as `0`. In the real world we'd look up what `null` means in the API, but here you can just do whatever you want.

**Comment by anott03** [comment]
#### In regards to the `null` issue:
To be thorough and make the typescript compiler happy, we need to check that every property on an object exists because we do not have a type for the `weather` object. So, for `weather.probabilityOfPrecipitation.value`, we need to know first that `weather` is not null. Then, once we know that, we need to ask "is `weather.probabilityOfPrecipitation`" null? And then the same thing for `weather.probabilityOfPrecipitation.value`.

The reason for this is that `weather` is type `any`, so typescript has no way of knowing what properties it should have and what the types of those properties should be. If this was going to be something we used in the long term, I would declare my own type along the lines of:

```typescript
type WeatherData {
    probabilityOfPrecipitation: {
        value: number
    }
}
```

**Review by anott03 — CHANGES_REQUESTED** [review]
Looks good! Just address the type issue and you should be all set. If you have questions about how you might go about doing that, I'm happy to provide some guidance.

**Review by anott03 — APPROVED** [review]
lovely

**Pull request merged**